src/test/java/com/google/sps/ProductPhotoDetectorTest.java

```diff
@@ -0,0 +1,140 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import com.google.sps.data.ProductDetectionData;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.jupiter.api.Assertions;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** 
+ * Tests the behaviour of ProductPhotoDetector, namely the shopping query returned and the 
+ * exceptions thrown.
+ */
+@RunWith(JUnit4.class)
+public final class ProductPhotoDetectorTest {
+
+  private static final byte[] IMAGE_BYTES = "Fake image".getBytes();
+  private ProductPhotoDetector productPhotoDetector;
+  private FakeProductDetectionAPI fakeProductDetectionAPI;
+
+  @Before
+  public void setUp() {
+    productPhotoDetector = new ProductPhotoDetector();
+    fakeProductDetectionAPI = new FakeProductDetectionAPI();
+  }
+
+  @Test
+  public void completeDetectionData() throws Exception {
+    List<String> labels = new ArrayList();
+    labels.add("Shoe");
+    labels.add("Sneaker");
+    List<String> logos = new ArrayList();
+    logos.add("Nike");
+    List<String> colors = new ArrayList();
+    colors.add("Black");
+
+    ProductDetectionData productDetectionData = new ProductDetectionData(labels, logos, colors);
+    fakeProductDetectionAPI.setReturnValue(productDetectionData);
```

Comment:
It will be nice if you could create a private method to do such setups,  e.g.

"setProductDetectionResult(ImmutableList<String> labels, ....)

By doing this:
#1 your unit test will be easier to read
#2 you can avoid boilerplate code among different tests.

Reply:
Indeed, solved!